src/psyclone/dynamo0p3.py

```diff
@@ -5830,18 +5836,18 @@ def _halo_read_access(self, arg):
                     "currently unsupported for kernels with stencil "
                     "accesses. Found '{0}'.".format(self._upper_bound_name))
             return self._upper_bound_name in ["cell_halo", "ncells"]
-        if arg.type in VALID_SCALAR_NAMES:
+        if arg.type in GH_VALID_SCALAR_NAMES:
```

Comment:
Please replace the `elif`s that are after `return`s with just `if`

Reply:
Done.